src/app/Scenes/MyCollection/Screens/ArtworkForm/Components/Rarity.tsx

```
@@ -6,13 +7,11 @@ import { Flex, Input, INPUT_HEIGHT, InputTitle, Separator, Text } from "palette"
 import { Select } from "palette/elements/Select"
 import React, { useState } from "react"
 
-export type AttributionClassType = "LIMITED_EDITION" | "OPEN_EDITION" | "UNIQUE" | "UNKNOWN_EDITION"
```

We can use the type generated by Relay instead.

---

**issue:** I would only do that if I was getting values from Gravity, but I think those values are being set in the frontend, I might be wrong though

---

This type actually matches the GraphQL type from MP/Gravity, so I don't see why we need to redefine it here.

---

When we use enums that are hardcoded in the client and have them typed using a dynamic type, we can end up having a type that is not necessarily matching the hardcoded values and it won't raise an error since the API is also expecting the same dynamic type. As I said above, I am assuming here that the rarity types are hardcoded in Eigen if it's not then you can ignore my comment

---

I see what you mean. The values should match the options of the Rarity select component and I'm ok with using the type from Relay or hardcoding them here. I'll bring `AttributionClassType` back :)

---

thank you Ole. let's speak about it if you would like to over zoom next week, maybe I am not getting something properly